### PR TITLE
[MODINREACH-173] INN-Reach Circulation Flow: React to Change in Request…

### PR DESCRIPTION
**Pull request by aleksandr2110**
… Record for INN-Reach Transaction: Cancel Request (Item Hold)

## Purpose
https://issues.folio.org/browse/MODINREACH-173

## Approach
1. mod-inn-reach should consume request-storage domain events via Kafka ([CIRCSTORE-302](https://issues.folio.org/browse/CIRCSTORE-302))
2. mod-inn-reach should check if the request is associated with an active INN-Reach Item Hold transaction
3. If so the then mod-inn-reach should issue a call to the owning site cancels request endpoint (D2IR API reference, pg. 58)
Use instance HRID as local bibId
Use patron name from item hold transaction record
4. Set item hold transaction status to CANCEL_REQUEST

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/test/java/org/folio/innreach/domain/listener/KafkaCirculationEventListenerApiTest.java [hunk 1]
```diff
@@ -276,6 +283,36 @@ void shouldSkipIfTransactionNotFoundByRequestId() {
...
 
+  @Test
+  @Sql(scripts = {
+    "classpath:db/central-server/pre-populate-central-server.sql",
+    "classpath:db/inn-reach-transaction/pre-populate-another-inn-reach-transaction.sql"
+  })
+  void shouldUpdateTransactionWhenCancelRequest() {
+    var folioRequestId = UUID.fromString("26278b3a-de32-4deb-b81b-896637b3dbeb");
+    var transactionId = UUID.fromString("2d219267-4e51-4843-b2c6-d9c44d313739");
+    var instanceId =  UUID.fromString("18fedcfa-81d4-4c29-a7ee-2540f57cca71");
```
Comment (oliinyko) [inline src/test/java/org/folio/innreach/domain/listener/KafkaCirculationEventListenerApiTest.java]: typically when we refer to pre populated data of test scripts we create constants for this
https://github.com/folio-org/mod-inn-reach/blob/1464b3971f01fc187cc653f01bb63d8369f2b8cd/src/test/java/org/folio/innreach/domain/listener/KafkaCirculationEventListenerApiTest.java#L74-L77

File: src/test/java/org/folio/innreach/domain/listener/KafkaCirculationEventListenerApiTest.java [hunk 2]
```diff
@@ -276,6 +283,36 @@ void shouldSkipIfTransactionNotFoundByRequestId() {
...
+
+    listener.handleRequestStorage(asSingleConsumerRecord(CIRC_REQUEST_TOPIC, folioRequestId, event));
+
+    verify(eventProcessor).process(anyList(), any(Consumer.class));
+    verify(instanceStorageClient, times(1)).getInstanceById(any());
+    verify(innReachExternalService, times(1)).postInnReachApi(any(), any(), any());
+    Mockito.verifyNoMoreInteractions(itemService);
+
+    var updatedTransaction = transactionRepository.fetchOneById(transactionId).orElse(null);
+    assertEquals(InnReachTransaction.TransactionState.CANCEL_REQUEST, updatedTransaction.getState());
```
Comment (oliinyko) [inline src/test/java/org/folio/innreach/domain/listener/KafkaCirculationEventListenerApiTest.java]: add static import for CANCEL_REQUEST

File: src/test/java/org/folio/innreach/domain/listener/KafkaCirculationEventListenerApiTest.java [hunk 3]
```diff
@@ -276,6 +283,36 @@ void shouldSkipIfTransactionNotFoundByRequestId() {
...
+    var request = event.getData().getNewEntity();
+    var instance = new Instance();
+    request.setId(folioRequestId);
+    request.setInstanceId(instanceId);
+    request.setStatus(CLOSED_CANCELLED);
+
+    when(instanceStorageClient.getInstanceById(request.getInstanceId())).thenReturn(instance);
+    when(innReachExternalService.postInnReachApi(any(), any(), any())).thenReturn("ok");
+
+    listener.handleRequestStorage(asSingleConsumerRecord(CIRC_REQUEST_TOPIC, folioRequestId, event));
```
Comment (oliinyko) [inline src/test/java/org/folio/innreach/domain/listener/KafkaCirculationEventListenerApiTest.java]: what is `handleRequestStorage` by the way? Lets rename it to `handleRequestEvents`

Comment (aleksandr2110) [reply src/test/java/org/folio/innreach/domain/listener/KafkaCirculationEventListenerApiTest.java]: Fixed

File: src/test/java/org/folio/innreach/domain/listener/KafkaCirculationEventListenerApiTest.java (hunk 2 above)
Comment (aleksandr2110) [reply src/test/java/org/folio/innreach/domain/listener/KafkaCirculationEventListenerApiTest.java]: Fixed

File: src/test/java/org/folio/innreach/domain/listener/KafkaCirculationEventListenerApiTest.java (hunk 1 above)
Comment (aleksandr2110) [reply src/test/java/org/folio/innreach/domain/listener/KafkaCirculationEventListenerApiTest.java]: Fixed

**Review by oliinyko — APPROVED** [review]

**Pull request merged**